src/v/pandaproxy/schema_registry/protobuf.cc

```diff
@@ -303,11 +307,61 @@ ss::future<const pb::FileDescriptor*> import_schema(
 struct protobuf_schema_definition::impl {
     pb::DescriptorPool _dp;
     const pb::FileDescriptor* fd{};
+
+    /**
+     * debug_string swaps the order of the import and package lines that
+     * DebugString produces, so that it conforms to
+     * https://protobuf.dev/programming-guides/style/#file-structure
+     *
+     * from:
+     * syntax
+     * imports
+     * package
+     * messages
+     *
+     * to:
+     * syntax
+     * package
+     * imports
+     * messages
+     */
+    ss::sstring debug_string() const {
+        auto s = fd->DebugString();
+
+        // reordering not required if no package or no dependencies
+        // if no syntax, then also early return
+        if (fd->package().empty() || fd->dependency_count() == 0) {
+            return s;
+        }
+
+        std::string_view sv{s};
+
+        constexpr size_t expected_syntax_len = 18;
+        auto syntax_pos = sv.find("syntax = \"proto");
+        auto syntax_end = syntax_pos + expected_syntax_len;
+
+        auto package = fmt::format("package {};", fd->package());
+        auto package_pos = sv.find(package);
+
+        auto imports_pos = syntax_end;
+        auto imports_len = package_pos - syntax_end;
+
+        auto trim = [](std::string_view sv) {
+            return boost::algorithm::trim_copy_if(
+              sv, [](char c) { return c == '\n'; });
+        };
+        auto header = trim(sv.substr(0, syntax_end));
+        auto imports = trim(sv.substr(imports_pos, imports_len));
+        auto footer = trim(sv.substr(package_pos + package.length()));
+
+        return ssx::sformat(
+          "{}\n{}\n\n{}\n\n{}\n", header, package, imports, footer);
+    }
```

Comment:
`footer` contains the package but we put package into the string at this line as well. So this seems like we writing the package twice. If so, should that happen?

Reply:
`footer` is after package (`sv.substr(package_pos + package.length()` is _from after_ package to end).

Reply:
Ohh right. I misread that line.